File: __tests__/fixtures/java/DeadCodeUnusedFields.java
package com.example;

public class DeadCodeUnusedFields {

    private String usedField = "hello";
    private int unusedField = 42;
    private static final long serialVersionUID = 1L;
    public String publicField = "pub";

    public void process() {
        System.out.println(usedField);
    }
}

File: __tests__/fixtures/kotlin/DeadCodeUnusedFields.kt
package com.example

class DeadCodeUnusedFields {

    private val usedField = "hello"
    private var unusedField = 42
    val publicField = "pub"

    fun process() {
        println(usedField)
    }
}

data class DataTest(val name: String, val age: Int)

class Delegated {
    private val lazyField by lazy { "value" }
}
